### DISCUSSION TITLE
Test Discussion

### DISCUSSION THREAD

**Discussion by mmliam** [micropython/micropython#9154] [General]
Standard format
`Line 1
Line 2
Line 3
`

After extra CR-LF inserted
`Line 1

Line 2

Line 3

`

**Comment by jimmo** [comment]
Please see my reply to your question here: https://forum.micropython.org/viewtopic.php?f=2&t=12982&p=70809#p70807

**Comment by mmliam** [comment]
As you can see the standard format ignored the CR-LF's after the lines.

**Reply by jimmo** [reply]
You need to use triple-backtick.

Your post is:

![Screenshot_2022-08-31_00-38-37](https://user-images.githubusercontent.com/5102768/187466326-65cf8180-ac73-4cca-8610-d9f0312d1a7c.png)

but it should be

![Screenshot_2022-08-31_00-39-00](https://user-images.githubusercontent.com/5102768/187466487-a5f8792f-dc5e-4688-ae10-d3a401027f45.png)

which would result in

![Screenshot_2022-08-31_00-39-46](https://user-images.githubusercontent.com/5102768/187466437-d127b575-f09f-47df-ad9d-c0261c9c4a5e.png)

**Reply by mmliam** [reply]
How do you explain the "standard format"? My source had a CRLF after lines 1, 2, & 3, but they were ignored. I had used the "Add Code, <Cntr-e>" Icon.

Repeating:
`Line 1
Line 2
Line 3
`
There are three small ticks before and after the code.

After added extra CRLF to source code:
`Line 1

Line 2

Line 3`

**Reply by jimmo** [reply]
I don't understand... this is what GitHub thinks you posted:

![Screenshot_2022-08-31_00-45-52](https://user-images.githubusercontent.com/5102768/187468008-1cfad635-f0a2-4050-b71e-ae9d6a9cb9c1.png)

**Reply by jimmo** [reply]
e.g. this works:

```
Line 1
Line 2
Line 3
```

![Screenshot_2022-08-31_00-48-05](https://user-images.githubusercontent.com/5102768/187468451-a70a791b-c739-46bd-8644-af4a11296356.png)

**Reply by mmliam** [reply]
Maybe there's some problem with the browser; I'm using FireFox.

**Reply by jimmo** [reply]
Testing

```
works
in
firefox?
```

**Reply by jimmo** [reply]
Maybe could you post a screenshot of what the editor looks like just before you click "reply"? What happens if you use the preview tab? Does it look correct there?

Possibly some browser extension you have installed?

**Comment by xytsrm** [comment]
Now responding in with Chrome.
`Line 1
Line 2
Line 3
`
Strange, I get the same thing in Chrome; Line's 1,2 & 3 appear on the same line.
Do you see my lines 1, 2, & 3 on the same line?
BTW: I created Lines 1,2,&3 in Notepad, so I know there were CRLF after each line.

**Reply by jimmo** [reply]
Can you post a screenshot of what the editor looks like just before you click "reply" ?

**Comment by xytsrm** [comment]
![GitHub Screenshot](https://user-images.githubusercontent.com/92377239/187471963-5e7fd61d-7718-4835-bb0b-2c52b18879f2.JPG)

Here's the screenshot of my last post.

**Reply by jimmo** [reply]
There is only a single backtick there.

**Reply by xytsrm** [reply]
Here's a screenshot of the preview of the same post.

![GitHub Screenshot (preview)](https://user-images.githubusercontent.com/92377239/187472919-c6b2695e-df1d-4088-8b07-3f7840a7f99e.JPG)

**Reply by xytsrm** [reply]
There are three very small ticks when I click on the <> Add code.
You can see from the screenshot of the preview the code is highlighted; I wouldn't get the highlighting if there wern't three ticks.

**Reply by jimmo** [reply]
In your screenshot you have

`` `Line 1 ``

but it should be

`` ```Line 1 ``

**Reply by jimmo** [reply]
GitHub has a difference between `print("inline snippets")` (using a single backtick) and

```py
print("multi")
print("line")
print("snippets")
```

using triple backticks.

You are getting highliting because you're using single backticks, but it's then making it into a single line because that's how inline snippets work.

**Reply by xytsrm** [reply]
How do you explain the highlighting?

**Reply by xytsrm** [reply]
I'm going to click on <>:
``
That's all I get.

**Reply by xytsrm** [reply]
All I see in the menu for code addition is <>, cursor over says "Add code". I don't see any mult-line option.

**Reply by jimmo** [reply]
You need to select the thing you want to highlight first.

When you have no selection, it assumes you want an inline snippet. So it's giving you the start and end backtick for the snippet (and it puts your cursor inside it). Hence why there are two backticks.

**Reply by mmliam** [reply]
I simply selected the lines from Notepad, as I would do from any IDE I happened to be using (e.g. Thonny).
Then pasted the between the ticks, as I do on the MicroPython site without any issues.

**Reply by robert-hh** [reply]
If it does not work with your IDE, just write the three backticks manually. That's what I'm doing, since I never got used to the <> icon.

**Reply by jimmo** [reply]
That's just not how it works.

Instead paste the code in. Select it. *Then* click <>.

Or, manually add the extra `` ` ``.

What used to be `[code]` or `[/code]` in the old site is now `` ``` ``.

**Comment by mmliam** [comment]
```
Line 1
Line 2
Line 3

```

Now I got it; wow.  You need to paste first, then highlight the pasted code, then select <>. Good grief, so different and complicated compared micropython site

**Reply by xytsrm** [reply]
Sorry, for all the confusion, on my part. But, on the MicroPython site I would click on the add-code, and simply paste between the XML - done.
So on GitHub I kept clicking on the add-code first, as I did on the MicroPython forum.

**Reply by mattytrentini** [reply]
Just for an alternative opinion; Github does it _exactly_ as I expect and it's _vastly_ easier than the old forum for me. And, of course, it has syntax highlighting which was sorely missed. Better in every way for me.

**Reply by stinos** [reply]
> So on GitHub I kept clicking on the add-code first, as I did on the MicroPython forum.

Once you're used to it, it should be faster when using just the keyboard: type the backticks (too bad ctrl-e is mapped to single backticks, not triple), hit ctrl-v, type closing backticks.

**Reply by robert-hh** [reply]
If you have highlighted a series of code lines, ctrl-e will enclose them with three backticks. So:

```
- paste the code
- highlight it
- push ctrl-e
```
